lighthouse-viewer/app/src/github-api.js

```diff
@@ -104,45 +100,46 @@ class GithubAPI {
                         'in to increase this limit.');
           }
 
-          const etag = resp.headers.get('ETag');
-
           if (!resp.ok) {
             if (resp.status === 304) {
               return cachedGist;
             } else if (resp.status === 404) {
               // Delete the entry from IDB if it no longer exists on the server.
-              idb.delete(id); // Note: async.
+              idbKeyval.delete(id); // Note: async.
             }
             throw new Error(`${resp.status} fetching gist`);
           }
 
+          const etag = resp.headers.get('ETag');
           return resp.json().then(json => {
-            const fileName = Object.keys(json.files)[0]; // Attempt to use first file in gist.
-            const f = json.files[fileName];
+            // Attempt to use first file in gist with report extension.
+            const filename = Object.keys(json.files)
+                .find(filename => filename.endsWith(GithubApi.LH_JSON_EXT));
+            if (!filename) {
+              throw new Error(`gist ${id} does not contain a Lighthouse report`);
+            }
+            const f = json.files[filename];
             if (f.truncated) {
-              return fetch(f.raw_url).then(resp => resp.json())
-                  .then(json => ({etag, content: json}));
+              return fetch(f.raw_url)
+                .then(resp => resp.json())
+                .then(content => ({etag, content}));
             }
             return {etag, content: JSON.parse(f.content)};
           });
         });
       });
+    }).then(response => {
+      // Cache the contents to speed up future lookups, even if an invalid
+      // report. Future requests for the id will either still be invalid or will
+      // not return a 304 and so will be overwritten.
+      return idbKeyval.set(id, response).then(_ => {
```

Comment:
> where was this caching done before?

this was originally in `lighthouse-report-viewer.js`, based on only caching it if it's a valid results object and can generate a LH report. I realized that doesn't matter, since if the user tries to reload the gist it's going to fetch the exact same invalid gist (unless github marks it changed), so might as well cache the response immediately. The benefit of the change is the idbKeyval logic for caching gist responses is now isolated in just this file